### PR TITLE
Enhancement | HTML tag in TreeModel value not showing properly.

### PR DESCRIPTION
**Pull request by Mohammed9531**

### CODE REVIEW

**Comment by Mohammed9531** [comment]
#57

**Comment by Mohammed9531** [comment]
Fix for HTML tag in TreeModel value not showing properly. #57

**Comment by Mohammed9531** [comment]
@rychkog Can you take a look at this PR as well and merge it?

**Comment by Mohammed9531** [comment]
@rychkog  Can you take a look at this PR as well and merge it?

Fix for HTML tag in TreeModel value not showing properly. #57

File: e2e/app.e2e-spec.ts [hunk 1]
```diff
@@ -23,4 +23,21 @@ describe('Tree App', () => {
...
+    page.navigateTo();
+
+    const firstChildNode = page.getFirstChildNode();
+    expect(browser.isElementPresent(firstChildNode)).toBeTruthy();
+    expect(firstChildNode.getText()).toEqual('Antiqua');
+
+    // verify if the element attributes are same
+    const attrs = {'id': 'antiqua', 'class': 'test'};
+
+    [].forEach.call(Object.keys(attrs), (key) => {
```
Comment (rychkog) [inline e2e/app.e2e-spec.ts]: `Object.keys(attrs).forEach` is a bit easier to read. Also you 100% safe in here calling `forEach` directly

File: e2e/app.e2e-spec.ts [hunk 2]
```diff
@@ -23,4 +23,21 @@ describe('Tree App', () => {
...
+      firstChildNode.getAttribute(key).then((value) => {
```
Comment (rychkog) [inline e2e/app.e2e-spec.ts]: This is executed via promises, hence code is async so It is better to use Promise.all for checking this.

```typescript
const expectations = Object.keys(attrs).map((key) => {
    return firstChildNode.getAttribute(key).then((value) => expect(value).toEqual(attrs[key]));
});

// And then just wait and assert for
Promise.all(expectations)
```

File: karma.conf.js [hunk 3]
```diff
@@ -29,6 +29,11 @@ module.exports = config => {
     webpack: require('./webpack.test.js'),
     webpackServer: {
       noInfo: true
-    }
+    },
+
+    // to avoid [PhantomJS 2.1.1 (Windows 8 0.0.0)]: Disconnected (1 times), because no message in 10000 ms error.
```
Comment (rychkog) [inline karma.conf.js]: You have the same change in your other request.

File: src/pipes/safe-html.pipe.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+
+@Pipe({ name: 'safeHtml'})
+export class SafeHtmlPipe implements PipeTransform  {
+  constructor(private sanitized: DomSanitizer) {}
+
+  transform(value:any) {
```
Comment (rychkog) [inline src/pipes/safe-html.pipe.ts]: It should accept value of type `string` and return value of type `SafeHtml`

File: src/pipes/safe-html.pipe.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+
+@Pipe({ name: 'safeHtml'})
+export class SafeHtmlPipe implements PipeTransform  {
```
Comment (rychkog) [inline src/pipes/safe-html.pipe.ts]: @Mohammed9531 We should cover this with its own unit test (I saw that you've added this to the `tree-internal.component.spec.ts`). SafeHtmlPipe should not depend on the tree.

File: test/tree-internal.component.spec.ts [hunk 6]
```diff
@@ -90,6 +92,7 @@ describe('TreeInternalComponent', () => {
     nodeMenuService = TestBed.get(NodeMenuService);
     nodeDraggableService = TestBed.get(NodeDraggableService);
     treeService = TestBed.get(TreeService);
+    saveHtml = TestBed.get(SafeHtmlPipe);
```
Comment (rychkog) [inline test/tree-internal.component.spec.ts]: saveHtml -> safeHtml

File: test/tree.spec.ts [hunk 7]
```diff
@@ -134,6 +134,33 @@ describe('Tree', () => {
...
+    };
+
+    const tree = new Tree(tags);
+    expect(tree).not.toBeFalsy('Constructed tree should exist');
+    expect(tree instanceof Tree).toBe(true, 'tree should be instance of Tree');
+    expect(tree.value).toEqual(tags.value);
+    expect(tree.children.length).toEqual(2);
+    expect(tree.children[0].value).toEqual(tags.children[0].value);
+
+    const next_sibling = tree.children[1];
```
Comment (rychkog) [inline test/tree.spec.ts]: camelCase - this is just a conventions to keep the code in the repo consistent.

**Comment by rychkog** [comment]
Also, there is a [guideline](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit) for commit messages. I need this in order to autogenerate changelog and release notes

**Comment by rychkog** [comment]
@Mohammed9531 I've added some tests and small fixed on top of your changes and cherry-picked them in https://github.com/valor-software/ng2-tree/pull/86

So I'm closing this PR, cause changes from it are already in `develop`

**Pull request closed without merging**